File: backend/app/shared/github_assets.py
from urllib.parse import urlparse


def is_github_asset_url(value: str) -> bool:
    try:
        url = urlparse(value)
        if url.scheme != "https":
            return False
        host = (url.hostname or "").lower()
        return host == "github.com" or host.endswith(".githubusercontent.com")
    except Exception:
        return False

File: web/src/app/shared/github-assets.ts
export function isGithubAssetUrl(value: string) {
  try {
    const url = new URL(value);
    if (url.protocol !== 'https:') return false;

    const host = url.hostname.toLowerCase();
    return host === 'github.com' || host.endsWith('.githubusercontent.com');
  } catch {
    return false;
  }
}
